modules/updater.py

```
@@ -0,0 +1,377 @@
+    """Queries the GitHub releases API on a background thread."""
+    update_available = pyqtSignal(str, str, str)  # (tag, html_url, asset_url)
+    up_to_date = pyqtSignal()
+    check_failed = pyqtSignal(str)
+
+    def run(self):
+        try:
+            url = f"https://api.github.com/repos/{_GITHUB_REPO}/releases/latest"
+            req = urllib.request.Request(
+                url,
+                headers={"Accept": "application/vnd.github+json", "User-Agent": "HoneyBatchr"},
+            )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)